src/utils/wallet/electrum.ts

```
@@ -257,48 +256,51 @@ export const subscribeToAddresses = async ({
 				let addressIndex =
 					currentWallet.addressIndex[selectedNetwork][addressType]?.index;
 				addressIndex = addressIndex > 0 ? addressIndex : 0;
-				const addressesToSubscribeTo = Object.values(addresses).filter(
-					(a) => Math.abs(a.index - addressIndex) <= GAP_LIMIT,
+
+				// Only subscribe up to the gap limit.
+				const addressesInRange = Object.values(addresses).filter(
+					(address) => Math.abs(address.index - addressIndex) <= GAP_LIMIT,
 				);
-				let i = 0;
-				for (const { scriptHash } of addressesToSubscribeTo) {
-					// Only subscribe up to the gap limit.
-					if (i > GAP_LIMIT) {
-						break;
-					}
-					scriptHashes.push(scriptHash);
-					i++;
-				}
+				const addressesToSubscribe = addressesInRange.slice(-GAP_LIMIT);
+				const addressScriptHashes = addressesToSubscribe.map(
+					({ scriptHash }) => scriptHash,
+				);
+
+				scriptHashes.push(...addressScriptHashes);
 			}
 		}
 	}
 
-	// Subscribe to all provided scriphashes.
-	await Promise.all(
-		scriptHashes.map(async (addressScriptHash) => {
-			const subscribeAddressResponse: ISubscribeToAddress =
-				await electrum.subscribeAddress({
-					scriptHash: addressScriptHash,
-					network: selectedNetwork,
-					onReceive: (): void => {
-						refreshWallet();
-						onReceive?.();
-					},
-				});
-			if (subscribeAddressResponse.error) {
-				return err('Unable to subscribe to receiving addresses.');
-			}
-		}),
-	);
+	// Subscribe to all provided script hashes.
+	const promises = scriptHashes.map(async (scriptHash) => {
+		const response: ISubscribeToAddress = await electrum.subscribeAddress({
+			scriptHash,
+			network: selectedNetwork,
+			onReceive: (): void => {
+				refreshWallet();
+				onReceive?.();
+			},
+		});
+		if (response.error) {
+			throw Error('Unable to subscribe to receiving addresses.');
+		}
+	});
+
+	try {
+		await Promise.all(promises);
+	} catch (e) {
+		console.log(e);
+		return err(e);
+	}
+
```

just refactoring here and avoiding swallowing errors